cli/internal/runsummary/spaces.go

```diff
@@ -69,36 +69,47 @@ func (c *spacesClient) start() {
 		return
 	}
 
-	mu := sync.Mutex{}
-	firstReqDone := false
-	processors := 8
-	for i := 0; i < processors; i++ {
-		c.wg.Add(1)
-		go func() {
-			defer c.wg.Done()
-			for req := range c.requests {
-				// since we have multiple processors, we want to lock firstReqDone so only the first one will mark firstReqDone
-				mu.Lock()
-				if !firstReqDone {
-					firstReqDone = true
-					mu.Unlock()
-
-					// make the first request and then close the run.created channel
-					// as a signal that other requests can proceed.
-					go func(r *spaceRequest) {
-						c.makeRequest(r)
-						close(c.run.created)
-					}(req)
-
-				} else {
-					mu.Unlock()
-
-					// If this is not the first request, wait for the run to be created
-					<-c.run.created
-					go c.makeRequest(req)
-				}
+	// Start an immediately invoked go routine that listens for requests coming in from a channel
+	pending := []*spaceRequest{}
+	firstRequestStarted := false
+
+	// Create a labeled statement so we can break out of the for loop more easily
+
+	// Setup a for loop that goes infinitely until we break out of it
+FirstRequest:
+	for {
+		// A select statement that can listen for messages from multiple channels
+		select {
+		// listen for new requests coming in
+		case req := <-c.requests:
+			if req == nil {
+				continue
```

Comment:
I added this check to prevent a bunch of `nil`s getting added to the `pending` slice, but I'm not sure why/how nil messages are being passed to this channel. My guess is it has something to do with the `for` loop, but I'm not sure.